src/v2i-hub/TimPlugin/src/TimPlugin.cpp

```diff
@@ -0,0 +1,438 @@
+#include "TimPlugin.h"
+#include <WGS84Point.h>
+#include "TimeHelper.h"
+#include "XmlCurveParser.h"
+#include "VehicleLocate.h"
+#include <tmx/messages/IvpDmsControlMsg.h>
+#include <tmx/j2735_messages/TravelerInformationMessage.hpp>
+
+using namespace std;
+using namespace tmx::messages;
+using namespace tmx::utils;
+
+namespace TimPlugin {
+
+
+
+
+/**
+ * Construct a new EpcwPlugin with the given name.
+ *
+ * @param name The name to give the plugin for identification purposes
+ */
+TimPlugin::TimPlugin(string name) :
+		PluginClient(name) {
+
+	AddMessageFilter < BsmMessage > (this, &TimPlugin::HandleBasicSafetyMessage);
+
+	SubscribeToMessages();
+
+}
+
+TimPlugin::~TimPlugin() {
+}
+
+void TimPlugin::UpdateConfigSettings() {
+
+	GetConfigValue<uint64_t>("Frequency", _frequency);
+
+	{
+		lock_guard<mutex> lock(_mapFileLock);
+		if (GetConfigValue<string>("MapFile", _mapFile))
+			_isMapFileNew = true;
+	}
+
+	GetConfigValue<uint64_t>("Snap Interval", _snapInterval);
```

Comment:
See my previous related comment to this and next line in manifest file.

Reply:
Addressed